circle.yml

```diff
@@ -1,23 +1,15 @@
 machine:
   ruby:
     version: 2.1.7
-  java:  # see <https://github.com/svenkreiss/html5validator#integration-with-circleci>
-    version: oraclejdk8
   environment:
-    GITHUB_PAGES_RUBY_VERSION: $(curl -s https://raw.githubusercontent.com/github/pages-gem/master/.ruby-version | cut -d '-' -f 1)
+    NOKOGIRI_USE_SYSTEM_LIBRARIES: true # Faster installation of nokogiri, required by html-proofer
 
 dependencies:
-  cache_directories:
-    - /home/ubuntu/.rvm/gems/
   post:
-    - if [[ ! -e /home/ubuntu/.rvm/gems/ruby-$GITHUB_PAGES_RUBY_VERSION/gems/github-pages-* ]]; then gem install github-pages --no-ri --no-rdoc; fi
-    - jekyll --version  # print it out
-    - sudo pip install html5validator
+    - bundle exec jekyll --version  # print it out
 
 test:
   override:
-    - grep version circle.yml | grep $GITHUB_PAGES_RUBY_VERSION  # Circle doesn't support interpolation of ruby version, so at least double-check we specified the proper one
-    - jekyll doctor
-    - jekyll build
-    - grep "css/main.css" _site/index.html  # smoke test that something was actually compiled
```

Comment:
I agree `jekyll` is supposed to exit with nonzero in case of failure. But this smoke test was added for a reason.

```sh
cd
mkdir tmp
cd tmp
jekyll build
echo $?
```

> If “test that something was actually compiled” isn't clear enough (as opposed to “compilation went well”), feel free to improve the comment.